### DISCUSSION TITLE
3 way diff support (todo)

### DISCUSSION THREAD

**Discussion by JBlond** [JBlond/php-diff#119] [General]
Isn't https://github.com/DigiLive/jQuery-Merge-for-php-diff the solution to 3 way diff, or has it to be done in this lib?

**Comment by DigiLive** [comment]
The repo you're referring to, is also a 2-way diff which uses the php-diff library to determine the differences and utilizes javascript to provide some kind of GUI.

3-way diff is a comparison between 3 files.
It would be a nice addition to this library, but I'm not quite sure this would take changes in the current code-base or if it will take additional code (classes) to execute multiple 2-way diffs between the 3 files and combine the results.

Lets say we have 3 files next to each other...
Instead of Added, Deleted, and Changed blocks (as in file comparison like the lib does now), there are now Matching, Non-Matching, and Conflict blocks.

- In a Matching block, the left and right files both differ from the center file, but are themselves equal. In other words, the same change from base was made in "theirs" and "yours".

- In a Non-Matching block, either the left or right file differs from the center file, but not both. In other words, the change was only made in one file.

- In a Conflict block, the left and right files both differ from the center files and are not equal to one other. In other words, different changes from base were made in "theirs" and "yours". Conflicts must be manually resolved by the user.
